### PR TITLE
Migration to Blockstack Connect

### PR DESCRIPTION
**Pull request by Filmaluco**
Updated Blockstack Login system, now uses connect instead of browser

### CODE REVIEW

File: assets/javascripts/pages/extension_login.js [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
 
 import Record from '../lib/records/record';
+import { showBlockstackConnect } from '@blockstack/connect';
 Record.config({ session: publicUserSession });
 
 function initAuthentication(loginBtn) {
   loginBtn.addEventListener('click', event => {
     event.preventDefault();
-    privateUserSession.redirectToSignIn(Constants.BLOCKSTACK_EXTENSION_REDIRECT_URI);
+    showBlockstackConnect(extensionAuthOptions);
```
Comment (Filmaluco) [inline assets/javascripts/pages/extension_login.js:13]: The resulting modal is not optimized to the current window size of the extension...

**Review by joaodiogocosta — CHANGES_REQUESTED** [review]
Hi @Filmaluco 👋 

There must be something wrong with the auth flow because it redirects me back to the extension path (`/extension_app`), and then to the landing page.

File: assets/javascripts/lib/blockstack_client.js [hunk 2]
```diff
@@ -8,3 +8,23 @@ function getUserSession(scopes) {
...
+    window.location = Constants.BLOCKSTACK_REDIRECT_URI;
+  },
+  userSession: privateUserSession,
+  appDetails: {
+    name: 'Envelop',
+    icon: 'https://envelop.app/images/manifest-icon.png',
+  },
+};
+
+const extensionOptions = options;
```
Comment (Filmaluco) [inline assets/javascripts/lib/blockstack_client.js:24]: @joaodiogocosta my JavaScript knowledge is limited, does this "clone" the object or is it a simple "reference"?
If this behaves like a reference that would explain the issue about the redirect to the extension...

Comment (joaodiogocosta) [reply assets/javascripts/lib/blockstack_client.js:24]: Yep, "reference":

image

To shallow copy an object you can use:

```js
const optionsA = { ... };
const optionsB = Object.assign({}; optionsA}; // ES5
const optionsB = { ...optionsA }; // ES6
```

To deep copy you will need a lib like [lodash](https://lodash.com/docs/#cloneDeep):
```js
const options A = { ... };
const optionsB = _.cloneDeep(optionsA);
```

Comment (Filmaluco) [reply assets/javascripts/lib/blockstack_client.js]: I did the shallow copy (ES5, not sure if I should have done the ES6...), let me know :)

**Comment by joaodiogocosta** [comment]
Closed by #97

**Comment by joaodiogocosta** [comment]
Closed by #97

**Pull request closed without merging**